packages/react-components/react-tree/library/src/components/TreeItem/useTreeItem.tsx

```diff
@@ -112,6 +105,15 @@ export function useTreeItem_unstable(props: TreeItemProps, ref: React.Ref<HTMLDi
       return;
     }
     const isEventFromExpandIcon = expandIconRef.current && elementContains(expandIconRef.current, event.target as Node);
+    if (!isEventFromExpandIcon) {
```

Comment:
according to the PR description there are several on clicks that should not result in the tree item's click handler being invoked.

From the changeset I only see that expand icon click is being considered here. Is that correct or am I missing smth?

Reply:
this might be a bit cleaner 

```ts
const handleClick = useEventCallback((event: React.MouseEvent<HTMLDivElement>) => {
    const isEventFromActions = actionsRef.current && elementContains(actionsRef.current, event.target as Node);
    const isEventFromSubtree = subtreeRef.current && elementContains(subtreeRef.current, event.target as Node);
    const isEventFromSelection = selectionRef.current && elementContains(selectionRef.current, event.target as Node);
    const isEventFromExpandIcon = expandIconRef.current && elementContains(expandIconRef.current, event.target as Node);

    if (
      isEventFromActions ||
      isEventFromSubtree ||
      isEventFromSelection ||
      event.isDefaultPrevented() ||
      itemType === 'leaf'
    ) {
      return;
    }

    if (!isEventFromExpandIcon) {
      onClick?.(event);
    }

    ReactDOM.unstable_batchedUpdates(() => {
      const data = {
        event,
        value,
        open: getNextOpen(),
        target: event.currentTarget,
        type: isEventFromExpandIcon ? treeDataTypes.ExpandIconClick : treeDataTypes.Click,
      } as const;
      props.onOpenChange?.(event, data);
      requestTreeResponse({
        ...data,
        itemType,
        requestType: 'open',
      });
      requestTreeResponse({
        ...data,
        itemType,
        parentValue,
        requestType: 'navigate',
        type: treeDataTypes.Click,
      });
    });
  });
```